src/clients/discord/discord.voice.service.ts

```
@@ -97,17 +102,43 @@ export class DiscordVoiceService {
       guildId: channel.guildId,
     });
 
+    this.voiceChannelId = channel.id;
+
     this.jellyfinWebSocketService.initializeAndConnect();
 
     if (this.voiceConnection === undefined) {
       this.voiceConnection = getVoiceConnection(member.guild.id);
     }
+
     this.voiceConnection?.on(VoiceConnectionStatus.Disconnected, () => {
       if (this.voiceConnection !== undefined) {
         this.playbackService.getPlaylistOrDefault().clear();
         this.disconnect();
       }
     });
+
+    const mem = member;
+    setTimeout(() => {
+      // leaving channel cuz no listener left
+      const interaval = setInterval(async () => {
+        if (!this.voiceChannelId) return;
+        const voiceChannel = (await mem.guild.channels.fetch(
+          this.voiceChannelId,
+        )) as VoiceChannel;
+        // this.logger.log(voiceChannel?.name);
+        if (!voiceChannel) return;
+        const members = voiceChannel.members.filter((s) => !s.user?.bot);
+        // this.logger.log(members.size);
+        if (members.size === 0) {
+          try {
+            this.stop(true);
+            this.disconnect();
+            clearInterval(interaval);
+          } catch (E) {}
```

Do not swallow exceptions, log as error